docs/devs/e2e-tests.md

```diff
@@ -171,20 +171,19 @@ test('Ordering a lab test for an OpenMRS patient creates the corresponding SENAI
 });
 
 test.afterEach(async ({ page }) => {
-  homePage = new HomePage(page);
   await homePage.deletePatient();
   await page.close();
 });
 ```
 
 We observe that the test structure is broken down between a **setup**, the actual **test case** and a **cleanup**:
 
-**Test Setup**: Before the actual test, we perform some preliminary actions: logging into Ozone (with SSO), creating a new patient, and starting a visit for the newly created patient.
+**Setup**: Instantiating a new 'homePage' object using the HomePage class constructor.
 
 **Test Case**: The core of each test case follows the _Given-When-Then_ pattern (organised here as Setup-Replay-Verification). We highly recommend this structured approach as it clearly delineates the setup of the test ("Setup"), the end-user actions performed ("Replay"), and the assertion of outcomes as experienced by the end-user ("Verification"). In our example:
 
-- **Setup**: Omitted here. All aspects of the setup have been performed in the `beforeEach()` method.
-- **Replay**: Navigation to the lab order form, add a lab test, and save the form.
-- **Verification**: Navigation to the SENAITE HIS component and search for the client by name. Verify that the client's name is visible in the clients list.
+- **Replay**: Logging into Ozone (with SSO), creating a new patient, and starting a visit for the newly created patient. Navigating to the lab order form, adding a lab test, and saving the form.
```

Comment:
Well, precisely I would assume that authenticating the user with SSO within Ozone would be done in `beforeEach()`, that's precisely something that you would want to do once for all your test cases no?

Reply:
We'd a discussion [here](https://github.com/ozone-his/ozone-e2e/pull/85#discussion_r1590610501) and agreed on making this refactoring.  So the E2E test cases are structured in the following format:
```
{
  // replay
  navigateToA();
  clickHereOnA();

  // verify
  navigateToB();
  readThereOnB()
}
```

Reply:
The source code is located at https://github.com/ozone-his/ozone-e2e/blob/main/e2e/tests/openmrs-senaite-flows.spec.ts#L12-L29

Reply:
Still, authenticating your test user into Ozone can be part of `beforeEach()`.

Let's try to be component-specific when performing actions. Couldn't you have a class of sorts to perform all OpenMRS-specific actions and end up encapsulating OpenMRS-specific logic in it? Like:
```javascript
openmrs.createPatient();
openmrs.goToLabOrderForm();
```
... etc.
This might not be possible, but then we need to come up with another naming system so that it's clearer which HIS component we're interacting with. Perhaps @Ruhanga could weigh in and tell me if I'm overcomplicating this?

Think of the day when we will bring Bahmni EMR in the mix.

Reply:
> This is on my radar to refactor actions performed in O3, Odoo, SENAITE and Keycloak into distinctive files. I expect to do this refactoring before this Thursday.

Logged https://mekomsolutions.atlassian.net/browse/OZ-562